### DISCUSSION TITLE
Search in hierarchal data

### DISCUSSION THREAD

**Discussion by ramax495** [OrchardCMS/OrchardCore#10758] [Q&A]
Hi there!
I am new in Orchard and want to know is there any tools to search through several levels of hierarchal data.
For example I have this structue:

```
Root
--Category1
----Category11
-------Product1 [Property=10]
-------Product2 [Property=20]
----Category12
-------Category121
----------Product3 [Property=10]
----------Product4 [Property=20]
```

And I need to construct such query that could select all nodes of type Product which have Property value = 10 inside Category1 from all sublevels of hierarchy.

Is there any API in Orchard Core that could make it with one query ?

**Comment by ns8482e** [comment]
I guess using taxonomy you can define tree structure https://docs.orchardcore.net/en/dev/docs/reference/modules/Taxonomies/
